### ISSUE TITLE
free memory drop down from 10G to 400M and system killed the training eventually 

### ISSUE THREAD

**Issue by ryantan1** [mozilla/TTS#471]
Hi,

I test the master branch[599c2db merge request 456] with http://data.keithito.com/data/speech/LJSpeech-1.1.tar.bz2
my environment:
ubuntu 1804
python 3.7.5 virtual env
memory 16G

1. follow the guide as https://gist.github.com/erogol/97516ad65b44dbddb8cd694953187c5b
    wget http://data.keithito.com/data/speech/LJSpeech-1.1.tar.bz2
   tar -xjf LJSpeech-1.1.tar.bz2
   shuf LJSpeech-1.1/metadata.csv > LJSpeech-1.1/metadata_shuf.csv
   head -n 12000 LJSpeech-1.1/metadata_shuf.csv > LJSpeech-1.1/metadata_train.csv
   tail -n 1100 LJSpeech-1.1/metadata_shuf.csv > LJSpeech-1.1/metadata_val.csv
   git clone https://github.com/mozilla/TTS
   sudo apt-get install espeak
   pip install soundfile
   cd TTS
   python setup.py install
----use following code to generate local config.json-----
   import json
   from utils.io import load_config
   CONFIG = load_config('config.json')
   CONFIG['datasets'][0]['path'] = '../LJSpeech-1.1/'
   CONFIG['output_path'] = '../'
   with open('config.json', 'w') as fp:
       json.dump(CONFIG, fp)
-----code end------------
2.  run python train.py --config_path config.json,
3. open top in new terminal, you will find the free memory drop down quickly when free memory reaches several hundreds MB,
    the train was killed by system.

Thanks for your investigation~

**Comment by ryantan1** [comment]
------training process in terminal----------

Using CUDA: False
Number of GPUs: 0
Git Hash: 599c2db
Experiment folder: ../LJSpeech/ljspeech-July-29-2020_09+26AM-599c2db
Setting up Audio Processor...
| > sample_rate:22050
| > num_mels:80
| > min_level_db:-100
| > frame_shift_ms:None
| > frame_length_ms:None
| > ref_level_db:20
| > num_freq:513
| > power:1.5
| > preemphasis:0.0
| > griffin_lim_iters:60
| > signal_norm:True
| > symmetric_norm:True
| > mel_fmin:0
| > mel_fmax:8000.0
| > max_norm:4.0
| > clip_norm:True
| > do_trim_silence:True
| > trim_db:60
| > do_sound_norm:False
| > stats_path:None
| > hop_length:256
| > win_length:1024
| > n_fft:1024
Using model: Tacotron2
| > Num output units : 513

Model has 28947858 parameters

EPOCH: 0/1000

Number of output frames: 7

DataLoader initialization
| > Use phonemes: False
| > Number of instances : 12000
| > Max length sequence: 187
| > Min length sequence: 5
| > Avg length sequence: 98.47641666666667
| > Num. instances discarded by max-min (max=153, min=6) seq limits: 444
| > Batch group size: 0.

TRAINING (2020-07-29 09:26:52)

--> STEP: 24/180 -- GLOBAL_STEP: 25
| > decoder_loss: 3.86587 (3.92852)
| > postnet_loss: 5.71722 (5.89929)
| > stopnet_loss: 0.73470 (0.76719)
| > ga_loss: 0.05319 (0.07255)
| > loss: 9.63628
| > align_error: 0.98838 (0.98719)
| > avg_spec_len: 336.53125
| > avg_text_len: 59.109375
| > step_time: 20.76
| > loader_time: 0.01
| > lr: 0.00010
killed

-----training end-------------
and LJSpeech, only two files there:
.
└── ljspeech-July-29-2020_09+26AM-599c2db
├── config.json
├── events.out.tfevents.1595986011.ubuntu
└── test_audios

2 directories, 2 files

**Comment by ryantan1** [comment]
checked the dev branch last commit[c5bda2 submitted on 28th July], Igot same problem with full ljspeech dataset

**Comment by erogol** [comment]
do you have a GPU on your machine ?

**Comment by ryantan1** [comment]
> do you have a GPU on your machine ?

erogol, on GPU on my machine

**Comment by erogol** [comment]
with no GPU training would take ages. FYI

You can follow this link to debug your issue.  pytorch/pytorch#8976

I don't think it is about TTS but PyTorch. Let me know your progress.

**Comment by ryantan1** [comment]
torch version: 1.5.1
set num_loader_workers to 0, same problem

**Comment by ryantan1** [comment]
@erogol, Itried to use objgraph to locate the memory leak issue, the strange thing is that there rarely no obj increases for some time  while free memory still drop down till killed.
 > TRAINING (2020-07-30 11:38:47) 
function                      71789    +71789
dict                          37198    +37198
tuple                         36974    +36974
list                          26338    +26338
cell                          18296    +18296
weakref                       13809    +13809
getset_descriptor              8454     +8454
type                           8220     +8220
builtin_function_or_method     6627     +6627
method_descriptor              5898     +5898
property                       5569     +5569
wrapper_descriptor             4485     +4485
module                         3910     +3910
ModuleSpec                     3906     +3906
SourceFileLoader               3585     +3585
staticmethod                   2997     +2997
TFDecorator                    2585     +2585
set                            2538     +2538
ArgSpec                        1345     +1345
classmethod                    1294     +1294
----format_data----- 8
Tensor                          183      +152
dict                          37260       +62
list                          26341        +3
member_descriptor               914        +2
wrapper_descriptor             4486        +1
builtin_function_or_method     6628        +1
method_descriptor              5899        +1
weakref                       13810        +1
----format_data----- 8
dict                37265        +5
Tensor                187        +4
tuple               36977        +3
set                  2540        +2
list                26342        +1
module               3911        +1
ModuleSpec           3907        +1
FileFinder            541        +1
SourceFileLoader     3586        +1
----format_data----- 8
----format_data----- 8
tuple    36978        +1
----format_data----- 8
----format_data----- 8
----format_data----- 8
----format_data----- 8
----format_data----- 8
----format_data----- 8
weakref                       13820       +10
builtin_function_or_method     6633        +5
dict                          37266        +1
set                            2541        +1
----format_data----- 8
Tensor      189        +2
----format_data----- 8
----format_data----- 8
----format_data----- 8
----format_data----- 8
----format_data----- 8
----format_data----- 8
----format_data----- 8
----format_data----- 8
----format_data----- 8
----format_data----- 8
----format_data----- 8
----format_data----- 8
----format_data----- 8
----format_data----- 8

   --> STEP: 24/180 -- GLOBAL_STEP: 25
     | > decoder_loss: 3.86587  (3.92852)
     | > postnet_loss: 5.71722  (5.89929)
     | > stopnet_loss: 0.73470  (0.76719)
     | > ga_loss: 0.05319  (0.07255)
     | > loss: 9.63628 
     | > align_error: 0.98838  (0.98719)
     | > avg_spec_len: 336.53125
     | > avg_text_len: 59.109375
     | > step_time: 41.96
     | > loader_time: 6.42
     | > lr: 0.00010
----format_data----- 8
----format_data----- 8
----format_data----- 8
----format_data----- 8
----format_data----- 8

Process finished with exit code 137 (interrupted by signal 9: SIGKILL)

**Comment by ryantan1** [comment]
and I add some code to checked whether it is caused by wav file un-closed after loading data.
from log, it seems it is NOT the case. anything more I can check, I am lost now

**Comment by ryantan1** [comment]
@erogol,could you please tell which python version are you using? you do NOT have such issue? Thanks~

**Comment by erogol** [comment]
I use python 3.7

**Comment by erogol** [comment]
Pls try the latest dev. It should have been fixed.
